### PR TITLE
test: reproducer @JsonValue and @JsonCreator issue

### PR DESCRIPTION
**Pull request by sdelamo**
see: https://github.com/micronaut-projects/micronaut-starter/pull/2192

### CODE REVIEW

**Comment by dstepanov** [comment]
Passing?

File: serde-tck-tests/src/main/java/io/micronaut/serde/tck/tests/JdkVersionTest.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    @Test
+    public void testJdkSerialization(JsonMapper jsonMapper) throws IOException {
+        //given:
+        JdkVersion jdk17 = JdkVersion.JDK_17;
+
+        //when:
+        String json = jsonMapper.writeValueAsString(new Foo(jdk17));
+
+        //then:
+        assertEquals("{\"jdk\":\"JDK_17\"}", json);
```
Comment (dstepanov) [inline serde-tck-tests/src/main/java/io/micronaut/serde/tck/tests/JdkVersionTest.java:38]: Use triple notation

**Comment by sdelamo** [comment]
> Passing?

it is passing because I excluded the test for serde. See `@ExcludeNamePatterns`

**Comment by dstepanov** [comment]
Can you make the included test fail?

**Comment by sdelamo** [comment]
sure, remove `@ExcludeNamePatterns` line in this PR and run `./gradlew :test-suite-tck-serde:test`. You will see `JdkVersionTest` fail

**Comment by dstepanov** [comment]
What's the point of including a new test if it doesn't fail? I'm confused what is `JdkVersionTest`?

**Comment by sdelamo** [comment]
> What's the point of including a new test if it doesn't fail? 

It does fail. I have pushed a [commit](https://github.com/micronaut-projects/micronaut-serialization/pull/709/commits/34db69498a73aec73537e2f415747a1a498f09da) which will make it fail in CI. 

> I'm confused what is `JdkVersionTest`?

[`JdkVersionTest` is the test included in this PR](https://github.com/micronaut-projects/micronaut-serialization/pull/709/files#diff-41e85fa366d590264896edc5884d41dd0b609f20085883ee0d7dc0158f3d3188).

**Comment by dstepanov** [comment]
Ah I see, I thought you disabled the test from the TCK and the new one is passing

**Comment by sdelamo** [comment]
It was closed via https://github.com/micronaut-projects/micronaut-serialization/pull/710/files#diff-eb4b206447172ff2236d03a097c747c107949925d8523eb7fd5808542f5eda74

**Pull request closed without merging**